client/my-sites/theme/main.jsx

```diff
@@ -644,9 +644,8 @@ export default connect(
 	( state, { id } ) => {
 		const selectedSite = getSelectedSite( state );
 		const siteSlug = selectedSite ? getSiteSlug( state, selectedSite.ID ) : '';
-		const isJetpack = selectedSite && isJetpackSite( state, selectedSite.ID );
-		const isWpcomTheme = selectedSite ? isThemeWpcom( state, id, selectedSite.ID ) : true;
-		const siteIdOrWpcom = ( isJetpack && ! isWpcomTheme ) ? selectedSite.ID : 'wpcom';
+		const isWpcomTheme = !! getTheme( state, 'wpcom', id );
+		const siteIdOrWpcom = ( selectedSite && ! isWpcomTheme ) ? selectedSite.ID : 'wpcom';
```

Comment:
This is a bit redundant: We're already using `getTheme` with `wpcom` in the `isWpcom` definition on the line right before this one, and now, we're getting it again if it's falsey. Maybe we can re-write this to illustrate the underlying pattern more clearly? Something like

```js
		let source = 'wpcom';
		let theme = getTheme( state, 'wpcom', id );
		const isWpcomTheme = !! theme;
		if ( selectedSite && ! isWpcomTheme ) {
			source = selectedSite.ID;
			theme = getTheme( state, selectedSite.ID, id );
		}
		const error = theme ? false : getThemeRequestErrors( state, id, source );
```

(at the bottom of this entire definitions block). Oh well, not sure that's any more elegant. More explicit, maybe.

`error` seems to be the hair in the soup here -- I for one can't seem to think of any nicer pattern to include that. So maybe the actual fix would be to not obtain `error` from a dedicated subtree, but instead use a boolean combination of `isRequesting` for both possible subtrees, and emptiness of themes.

Feel free to use or discard ^ :smile: